torch/csrc/jit/serialization/export_module.cpp

```diff
@@ -109,7 +132,10 @@ std::string getModuleTypeName(const Module& module, const std::string& prefix) {
 std::pair<IValue, c10::optional<IValue>> getFunctionTuple(
     const Module& module,
     const Function& func,
-    bool save_mobile_debug_info) {
+    const std::
+        unordered_map<at::Tensor, int, tensor_value_hash, tensor_value_equal>&
```

Comment:
Do we need to create hash by converting tensor to string? Can we not use underlying storage ptr? I dont know if we have aliasing in constant where two constant tensors might refer to the same underlying memory but we can safeguard against that by using tensor sizes on top of storage ptr.

Reply:
Hmm I don't think I use hash with ptr. The customized hash `tensor_value_hash` is using string and comes from here (https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/core/Formatting.cpp#L230). Here is one example:
```
     torch._utils._rebuild_tensor_v2(pers.obj(('storage', torch.FloatStorage, '0', 'cpu', 90944),),
       0,
       (1, 116, 28, 28),
       (90944, 784, 28, 1),
       False,
       collections.OrderedDict())
```
Do I misunderstand anything?

Reply:
@kimishpatel do you have any other concern for this change?

Reply:
Sorry on PTO today. What I meant to say is: Why are we converting tensor to str and using that as hash?
What are the other options? Can we not just do `std::hash<void*>(tensor.data_ptr())`? or mix tensor.data_ptr() with tensor.sizes()? Stringifying tensor for hashing seems like overkill.

Reply:
Oops sorry. I didn't notice you are on PTO today. It's actually intended to use string as the hash. The reason is that the key of `constants_from_jit` is tensor, and it's from the constants from jit, which has different ptr than the constants from mobile. I used the IValue hash function at the beginning, which uses ptr as hash for tensor type under the hood, instead of this customized hash function. It didn't work so I switched to string as hash input.


Side note: IValue hash function is defined here (https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/core/ivalue.cpp#L323-L334). For tensor type, ptr is part of the hash input.

Reply:
OK sounds good.